jobfunnel/backend/jobfunnel.py

```diff
@@ -0,0 +1,507 @@
+"""Scrapes jobs, applies search filters and writes pickles to master list
+Paul McInnis 2020
+"""
+import csv
+import json
+import logging
+import os
+import pickle
+import sys
+from concurrent.futures import ThreadPoolExecutor
+from datetime import date, datetime, timedelta
+from time import time
+from typing import Dict, List, Optional, Tuple
+
+from requests import Session
+
+from jobfunnel.backend import Job
+from jobfunnel.backend.tools import Logger
+from jobfunnel.backend.tools.filters import DuplicatedJob, JobFilter
+from jobfunnel.config import JobFunnelConfigManager
+from jobfunnel.resources import (CSV_HEADER, MAX_BLOCK_LIST_DESC_CHARS,
+                                 MAX_CPU_WORKERS,
+                                 MIN_JOBS_TO_PERFORM_SIMILARITY_SEARCH, T_NOW,
+                                 DuplicateType, JobStatus, Locale)
+
+
+class JobFunnel(Logger):
+    """Class that initializes a Scraper and scrapes a website to get jobs
+
+    NOTE: This is intended to be used with persistant cache and CSV files
+          dedicated to a single, consistant job search.
+    TODO: instead of Dic[str, Job] we should be using JobsDict
+    """
+
+    def __init__(self, config: JobFunnelConfigManager) -> None:
+        """Initialize a JobFunnel object, with a JobFunnel Config
+
+        Args:
+            config (JobFunnelConfigManager): config object containing paths etc.
+        """
+        super().__init__(
+            level=config.log_level,
+            file_path=config.log_file,
```

Comment:
If a _settings.yaml_ file is specified, the _log_file_ variable still uses the default path(~/job_search_results/). If the path doesn't exist, the initialization of the super class breaks.

Reply:
:+1: matches the issue with locale as well, will fix the CLI.

Reply:
ok this should have been resolved, I've added special handling for log_file